db/migrate/20160324203119_add_id_primary_key_to_join_tables.rb

```diff
@@ -0,0 +1,82 @@
+require_relative './20160106214719_add_composite_primary_keys_to_join_tables'
+
+class AddIdPrimaryKeyToJoinTables < ActiveRecord::Migration[5.0]
+  JOIN_TABLES = %w(
+    conditions_miq_policies
+    key_pairs_vms
+    miq_roles_features
+    miq_servers_product_updates
+    security_groups_vms
+    storages_vms_and_templates
+    miq_groups_users
+    cloud_tenants_vms
+    customization_scripts_operating_system_flavors
+    configuration_locations_configuration_profiles
+    configuration_organizations_configuration_profiles
+    configuration_profiles_configuration_tags
+    configuration_tags_configured_systems
+    container_groups_container_services
+    direct_configuration_profiles_configuration_tags
+    direct_configuration_tags_configured_systems
+    network_ports_security_groups
+  ).freeze
+
+  class MiqRegion < ApplicationRecord
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    say_with_time("Removing composite primary keys from join tables") do
+      revert AddCompositePrimaryKeysToJoinTables
+    end
+
+    JOIN_TABLES.each do |t|
+      delete_remote_region_rows(t) if on_replication_target?
+
+      say_with_time("Add primary key \"id\" to #{t}") do
+        execute <<-SQL
+          CREATE SEQUENCE #{sequence_name(t)} START #{seq_start_value}
+        SQL
+
+        execute <<-SQL
+          ALTER TABLE #{t} ADD COLUMN id BIGINT PRIMARY KEY
+          DEFAULT NEXTVAL('#{sequence_name(t)}')
+        SQL
+
+        execute <<-SQL
+          ALTER SEQUENCE #{sequence_name(t)} OWNED BY #{t}.id
+        SQL
+      end
+    end
+  end
+
+  def down
+    JOIN_TABLES.each do |t|
+      remove_column t.to_sym, :id
+    end
+
+    run AddCompositePrimaryKeysToJoinTables
+  end
+
+  def delete_remote_region_rows(table)
+    execute("SELECT * FROM #{table}").each do |row|
+      next if ApplicationRecord.id_in_current_region?(row.values.first)
+
+      conditions = row.map { |k, v| "#{k} = #{v}" }.join(" AND ")
+      execute "DELETE FROM #{table} WHERE #{conditions}"
+    end
+  end
+
+  def on_replication_target?
+    MiqRegion.select(:region).distinct.count > 1
```

Comment:
I was basing this mostly on https://github.com/ManageIQ/manageiq/blob/master/db/migrate/20151208150956_fix_host_storage_replication_on_upgrade.rb

When else would we have multiple region rows other than being a replication global region?
Also this is meant to indicate we are on the global region, so target as in where we are writing the rows to.

Hope that clears things up.